### PR TITLE
Display nationality as "Unknown" when it is not present in person definition

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

Jira ticket: https://dsdmoj.atlassian.net/browse/CAS2-236

# Changes in this PR

As above. Arguably there are other person fields that we should do the same with, but I've limited this to nationality based on what has been flagged.

### After

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/utils/checkYourAnswersUtils.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export const getApplicantDetails = (application: Application | Cas2SubmittedAppl
         text: 'Nationality',
       },
       value: {
-        html: nationality,
+        html: nationality || 'Unknown',
```
Comment (tahb) [inline server/utils/checkYourAnswersUtils.ts:246]: Should we add a unit test for this? I went looking but couldn't find this `getApplicantDetails` method tested at all in `checkYourAnswersUtils.test.ts`, so I suspect we may be able to create a prior commit that adds tests for the previous behaviour?

File: server/views/people/partials/personDetails.njk [hunk 2]
```diff
@@ -22,7 +22,7 @@
         text: "Nationality"
       },
       value: {
-        text: person.nationality
+        text: person.nationality or 'Unknown'
```
Comment (Harriethw) [inline server/views/people/partials/personDetails.njk:25]: do we need this here if 'Unknown' has already been set in the javascript function?

Comment (patrickjfl) [reply server/views/people/partials/personDetails.njk:25]: yes, this one is to populate the 'Confirm applicant details' page, the other one is to populate CYA and submitted application views

File: server/utils/checkYourAnswersUtils.test.ts [hunk 3]
```diff
@@ -517,4 +519,81 @@ describe('getPage', () => {
       getPage('confirm-eligibility', 'bar')
     }).toThrow(UnknownPageError)
   })
+
+  describe('getApplicantDetails', () => {
```
Comment (tahb) [inline server/utils/checkYourAnswersUtils.test.ts:523]: Now we have some test coverage I was thinking we could add a new shorter test for checking the default behaviour

```
person = personFactory.build({nationality: nil})
expected = {
          key: {
            text: 'Nationality',
          },
          value: {
            html: "Unknown",
          },
        },
 expect(getApplicantDetails(application)).toContain(expected)
```

Comment (patrickjfl) [reply server/utils/checkYourAnswersUtils.test.ts:523]: we could, but how much value is there in that test? Are we not at that point basically testing the javascript `||` operator?

Comment (tahb) [reply server/utils/checkYourAnswersUtils.test.ts:523]: The implementation is simple in this case but so is the test and having a test would protect it from regression and document the behaviour.

Comment (patrickjfl) [reply server/utils/checkYourAnswersUtils.test.ts:560]: OK yeah, good points. I have done this 👍

**Review by tahb — APPROVED** [review]

**Pull request merged**